### PR TITLE
feat: NM re-positioning preview

### PR DESCRIPTION
**Pull request by anxobotana**
Process the sorting for the UI until the API responds and confirms the sorting.

### CODE REVIEW

File: packages/app-bridge/src/react/useCategorizedDocumentPages.ts [hunk 1]
```diff
@@ -69,16 +83,53 @@ export const useCategorizedDocumentPages = (
...
+    newPosition: DocumentPagesMoveEvent['position'],
+) => {
+    if (!documentPage.sort || !newPosition) {
+        return documentPages;
+    }
+
+    const previousPosition = documentPage.sort;
+    const documentPagesAsArray: DocumentPage[] = [...documentPages.values()];
+
+    documentPages.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useCategorizedDocumentPages.ts:110]: Shouldn't we create a new map and avoid mutation here?

File: packages/app-bridge/src/react/useDocumentCategories.ts [hunk 2]
```diff
@@ -72,16 +85,53 @@ export const useDocumentCategories = (
...
+    newPosition: DocumentCategoryMoveEvent['position'],
+) => {
+    if (!documentCategory.sort || !newPosition) {
+        return documentCategories;
+    }
+
+    const previousPosition = documentCategory.sort;
+    const documentCategoriessAsArray: DocumentCategory[] = [...documentCategories.values()];
+
+    documentCategories.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useDocumentCategories.ts:112]: Same here for avoid mutation

File: packages/app-bridge/src/react/useDocumentGroups.ts [hunk 3]
```diff
@@ -46,31 +48,109 @@ export const useDocumentGroups = (appBridge: AppBridgeBlock | AppBridgeTheme, op
...
+    newGroupId: DocumentMoveEvent['newGroupId'],
+) => {
+    if (newGroupId || !document.sort) {
+        return documentGroups;
+    }
+
+    const previousPosition = document.sort;
+    const documentGroupsAsArray: DocumentGroup[] = [...documentGroups.values()];
+
+    documentGroups.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useDocumentGroups.ts:101]: And here

File: packages/app-bridge/src/react/useDocumentGroups.ts [hunk 4]
```diff
@@ -46,31 +48,109 @@ export const useDocumentGroups = (appBridge: AppBridgeBlock | AppBridgeTheme, op
...
+    newPosition: DocumentGroupMoveEvent['position'],
+) => {
+    if (!documentGroup.sort) {
+        return documentGroups;
+    }
+
+    const previousPosition = documentGroup.sort;
+    const documentGroupsAsArray: DocumentGroup[] = [...documentGroups.values()];
+
+    documentGroups.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useDocumentGroups.ts:131]: Here too

File: packages/app-bridge/src/react/useGroupedDocuments.ts [hunk 5]
```diff
@@ -89,18 +100,62 @@ export const useGroupedDocuments = (
...
+    newGroupId: DocumentMoveEvent['newGroupId'],
+) => {
+    if (!document.sort) {
+        return documents;
+    }
+
+    const previousPosition = document.sort;
+    const documentsAsArray: Document[] = [...documents.values()];
+
+    documents.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useGroupedDocuments.ts:130]: Here

File: packages/app-bridge/src/react/useUncategorizedDocumentPages.ts [hunk 6]
```diff
@@ -69,16 +83,54 @@ export const useUncategorizedDocumentPages = (
...
+    newPosition: DocumentPagesMoveEvent['position'],
+) => {
+    if (!documentPage.sort || !newPosition) {
+        return documentPages;
+    }
+
+    const previousPosition = documentPage.sort;
+    const documentPagesAsArray: DocumentPage[] = [...documentPages.values()];
+
+    documentPages.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useUncategorizedDocumentPages.ts:111]: Here

File: packages/app-bridge/src/react/useUngroupedDocuments.ts [hunk 7]
```diff
@@ -97,19 +111,82 @@ export const useUngroupedDocuments = (
...
+    newPosition: DocumentMoveEvent['position'],
+) => {
+    if (!document.sort || !newPosition) {
+        return documents;
+    }
+
+    const previousPosition = document.sort;
+    const documentsAsArray: Document[] = [...documents.values()];
+
+    documents.clear();
```
Comment (fabriciomendonca) [inline packages/app-bridge/src/react/useUngroupedDocuments.ts:143]: Here

File: packages/app-bridge/src/react/useCategorizedDocumentPages.ts (hunk 1 above)
Comment (anxobotana) [reply packages/app-bridge/src/react/useCategorizedDocumentPages.ts:110]: We use `immer`'s `produce` function for calling all these methods, that takes care of this and prevents mutations.

**Review by fabriciomendonca — APPROVED** [review]

**Pull request merged**